### PR TITLE
Added more factorial

### PR DESCRIPTION
**Pull request by christianbender**
* I added a recursively version (C++)   
* I fixed a bug by the javascript version and added test cases with ```console.log(...)```

### CODE REVIEW

File: Factorial/Javascript/Factorial.js [hunk 1]
```diff
@@ -4,4 +4,6 @@ function factorial(num) {
   return num * factorial(num-1);
 }
 
+// Test
+console.log(factorial(0));
```
Comment (AustinTice) [inline Factorial/Javascript/Factorial.js]: If we're going to add test cases be sure to comment what the expected output should be

**Comment by christianbender** [comment]
@AustinTice 
Done. I added a new file. Renamed the files. And changed the formatting.

**Comment by AustinTice** [comment]
Please edit this code according to the previous PR's style guidelines! Great work though! 👍

**Comment by christianbender** [comment]
@AustinTice Done. I changed the formatting according to previous C++ PR's. greetings

**Comment by christianbender** [comment]
@AustinTice I have done the changes. 
greetings Christian

**Pull request merged**